pkg/apis/experimental/types.go

```diff
@@ -508,35 +511,70 @@ type IngressStatus struct {
 	LoadBalancer api.LoadBalancerStatus `json:"loadBalancer,omitempty"`
 }
 
-// IngressRule represents the rules mapping the paths under a specified host to the related backend services.
-type IngressRule struct {
-	// Host is the fully qualified domain name of a network host, or its IP
-	// address as a set of four decimal digit groups separated by ".".
-	// Conforms to RFC 1738.
+// IngressHost represents the rules mapping the paths under a specified host to
+// the related backend services.
+type IngressHost struct {
+	// Host is the fully qualified domain name of a network host, as defined
```

Comment:
This is a long-winded way of saying it is a DNS_SUBDOMAIN, which we have specced elsewhere.